File: chia-rest/app/entities/backup.py
"""Blockchain backup related entity."""

from dataclasses import dataclass
from datetime import datetime

from app.entities.base import BaseEntity


@dataclass(kw_only=True)
class Backup(BaseEntity):
    """Data class representation of chia database backup file."""

    name: str
    size: int
    height: int | None = None
    created: datetime

    def as_dict(self):
        result = super().as_dict()
        result['created'] = self.created.isoformat()
        return result


@dataclass(kw_only=True)
class BackupProgress(BaseEntity):
    """Data class representation of backup progress from a given timestamp."""

    progress: float | None = None
    current_size: int | None = None
    goal_size: int | None = None

File: ChiaMonitor/app/src/main/java/com/example/chiamonitor/domain/Backup.kt
package com.example.chiamonitor.domain

import java.time.LocalDateTime

data class Backup(
    val name: String,
    val size: Long,
    val height: Int,
    val created: LocalDateTime,
)


data class BackupProgress(
    val progress: Double,
    val currentSize: Long,
    val goalSize: Long,
)